[CS2113-W12-3] WheresMyMoney

Hackin7 opened the pull request:
WheresMyMoney helps frugal and tech-savvy university students track how much they are spending and what they are spending it on. The application can provide summaries and statistical insights on their spending habits, optimised for people who prefer a CLI

okkhoy commented on src/main/java/wheresmymoney/Expense.java:
```diff
@@ -0,0 +1,49 @@
+    }
+
+    
+    public Float getPrice() {
+        return price;
+    }
+    public String getDescription() {
+        return description;
+    }
+    public String getCategory() {
+        return category;
+    }
+    public void setPrice(Float price) throws WheresMyMoneyException {
+        if (price == null) {
+            throw new WheresMyMoneyException("Expense's price shouldn't be null.");
+        }
+        this.price = price;
+        assert this.price != null : "Expense's price shouldn't be null.";
+    }
+    public void setDescription(String description) throws WheresMyMoneyException {
```
include some blank lines in between the methods for better readability

okkhoy commented on src/main/java/wheresmymoney/ExpenseList.java:
```diff
@@ -0,0 +1,172 @@
+
+
+    /**
+     * Add an expense to the end of the list
```
doesn't follow our guidelines for header comments

okkhoy commented on src/main/java/wheresmymoney/ExpenseList.java:
```diff
@@ -0,0 +1,172 @@
+                addExpense(Float.parseFloat(line[2]), line[1], line[0]);
+            }
+
+            // closing writer connection
```
It's a redundant comment and also looks incorrect.
Comment mentions writer; code references reader!

okkhoy commented on src/main/java/wheresmymoney/command/Command.java:
```diff
@@ -0,0 +1,18 @@
+
+import java.util.HashMap;
+
+public abstract class Command {
```
this and its child classes can do better with some header comments

okkhoy reviewed (commented):
Please pay attention to coding standards. 
Many non-trivial methods don't have header comments

KSanjith commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,323 @@
+The abstract Command class has been implemented to introduce an additional layer of abstraction between I/O and command execution, allowing for separation of handling command keywords and executing commands.
+
+
+<u>Implementation Details</u>
+
+The following diagram is an inheritance diagram for Command and its children classes. 
+This has been heavily simplified and only shows the key commands.
+
+![CommandInheritance.png](diagrams%2Fimages%2FCommandInheritance.png)
+
```
Given how simplified it is and how few details it contains, is this diagram really necessary? Feel like you could either add more details or remove this diagram.

KSanjith commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,323 @@
+
+![CommandInheritance.png](diagrams%2Fimages%2FCommandInheritance.png)
+
+
+The following diagram is a sequence diagram for execution of Command.
+
+![CommandExecutionSequence.png](diagrams%2Fimages%2FCommandExecutionSequence.png)
+
```
Is the execute method missing a return arrow and value?

KSanjith replied:
The return arrow from isExit() method is labelled as "boolean", which is a variable type. Should it instead be "exit :boolean" or "exit" or something similar, instead of just "boolean"

KSanjith commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,323 @@
+UML class diagram to show the dependency between `Command` and `ExpenseList` classes:
+
+![CommandAndExpenseList.png](diagrams%2Fimages%2FCommandAndExpenseList.png)
+
+The following diagram is a UML class diagram for `Expense` and `ExpenseList`:
+
+![ExpenseAndExpenseList.png](diagrams%2Fimages%2FExpenseAndExpenseList.png "UML Class Diagram for Expense and ExpenseList")
+
```
Should multiplicity be 0..1 and not 0.1 ?

KSanjith commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,323 @@
+To minimise the amount of checks that need to be done, the recurring expenses are only added after the user calls the `load` command.
+
+<u>Implementation Details</u>
+
+Below is the UML class diagram for `RecurringExpense` and `RecurringExpenseList`:
+
+![RecurringExpenseAndRecurringExpenseList.png](diagrams%2Fimages%2FRecurringExpenseAndRecurringExpenseList.png "UML Class Diagram for RecurringExpense and RecurringExpenseList class")
+
```
Should multiplicity be 0..1 and not 0.1 ?

ayushi0803 commented on docs/DeveloperGuide.md:
<img width="1470" alt="Screenshot 2024-10-30 at 13 23 46" src="https://github.com/user-attachments/assets/30aa81b6-5692-43bf-8393-d605910a3a7f">
can you make the definitino of 'expense' classes more specific? Is expense an overarching umbrella over ExpenseList and ExpenseFilter? Can you make it more specfific with an appended suffix perhaps?

ayushi0803 replied:
what does the command class do? can it be described further? maybe you can make categories of classes and define each class under the relevant category to ensure a seamless explanation
<img width="1470" alt="Screenshot 2024-10-30 at 13 26 52" src="https://github.com/user-attachments/assets/47618dd4-06b8-42f0-818c-12f07f5b346f">

ayushi0803 replied:
<img width="1470" alt="Screenshot 2024-10-30 at 13 32 22" src="https://github.com/user-attachments/assets/c70fa45b-0f3f-425d-905e-bd8dbebe252e">

i cant really see the 1? can it be rewritten as something else? or can it be highlighted to make it more visible because it looks like its a dotted line

ayushi0803 replied:
could you use colours for class diagrams to increase readability? perhaps the line spacing can be increased too?

ayushi0803 replied:
<img width="1470" alt="Screenshot 2024-10-30 at 13 39 42" src="https://github.com/user-attachments/assets/7b9d5ad1-1cc1-4919-8676-4f6c5868bc5d">
why are these lines of code overlapping the vertical box on the left side of the diagram 
eg - [while line = csvReader.readNext)!= null]
[for each RecurringExpense]
etc

ayushi0803 replied:
overall, a commendable effort that needs a little more work and you'll be there! good luck team x

KSanjith commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,323 @@
+![RecurringExpenseAndRecurringExpenseList.png](diagrams%2Fimages%2FRecurringExpenseAndRecurringExpenseList.png "UML Class Diagram for RecurringExpense and RecurringExpenseList class")
+
+Below is the sequence diagram for when the user calls the `load` command.
+
+![RecurringExpenseLoadFromCsvSequence.png](diagrams%2Fimages%2FRecurringExpenseLoadFromCsvSequence.png "UML Sequence Diagram for calling load command")
+
```
When instantiating new objects for File, FileReader and CSVReader, should the arrow point to the side of the object box, with an activation box coming right out of the object box to represent the constructor? And maybe you could perhaps not include the "new" keyword in the arrow? Can refer to https://nus-cs2113-ay2425s1.github.io/website/schedule/week10/topics.html#W10-1 for the proper formatting for object creation in sequence diagrams.

Sukkaito commented on docs/diagrams/images/ExpenseAndExpenseList.png:
I don't think constructors should return any value, as constructors don't really return anything 🤔

Sukkaito replied:
![image](https://github.com/user-attachments/assets/8a0bafe7-b33f-4380-940e-486f426614e1)
Also, should this be a "0..1"? (2 dots)

Sukkaito commented on docs/diagrams/images/RecurringExpenseAndRecurringExpenseList.png:
Same issue here, constructors having return value. Shouldn't this be the case?

Sukkaito replied:
![image](https://github.com/user-attachments/assets/6dadb947-6cef-4256-b555-f19c7ab69a33)
Same issue here, should this be "0..1" instead?

Sukkaito commented on docs/diagrams/images/RecurringExpenseLoadFromCsvSequence.png:
![image](https://github.com/user-attachments/assets/cc06912f-46f4-456e-8cf7-4bcdf4f5eb66)
Should this be an `opt` block, because there is no "else" block in this case?

KSanjith reviewed (commented):
Almost there

ayushi0803 reviewed (commented):
overall, a commendable effort that needs a little more work and you'll be there! good luck team x

Sukkaito reviewed (commented):
LGTM! I noted some possible changes that you guys might want to apply. Keep the good work! 👍

C2linaung commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,323 @@
+2. `Expense`, `ExpenseList`, `ExpenseFilter` classes: Model expenses that commands can interact with.
+3. `Storage` class: Stores information between sessions.
+4. Logger and other utility classes: Provide extra functionalities for the software.
+### UI and Parser
```
the indentation is off. The formatting could be better

C2linaung commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,323 @@
+
+The following diagram is a sequence diagram for execution of Command.
+
+![CommandExecutionSequence.png](diagrams%2Fimages%2FCommandExecutionSequence.png)
```
I think the diagram can be a bit confusing since i have to navigate both in the left and right direction. it might be better to should all the flows towards 1 direction
![image](https://github.com/user-attachments/assets/74af0473-a37c-44b9-9520-e311ae99c6eb)

C2linaung commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,323 @@
+
+![ExpenseAndExpenseList.png](diagrams%2Fimages%2FExpenseAndExpenseList.png "UML Class Diagram for Expense and ExpenseList")
+
+### Expense Filter
```
would it be better to have a sequence diagram to showcase each of the commands and how they work

C2linaung commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,323 @@
+
+Commands interact with `UI` and `Parser` classes via `Main`, as illustrated in the following class diagram:
+
+![UiToCommand.png](diagrams%2Fimages%2FUiToCommand.png)
```
This has alot of details. Do you think we can make it more concise

AjayShanker-geek commented on docs/UserGuide.md:
```diff
@@ -1,42 +1,130 @@
+- [Features](#features-)
+- [FAQ](#faq)
+- [Others](#others)
+- [Command Summary](#command-summary)
```
Great that you added Table of Content (TOC) to improve readability of the Developer Guide.

AjayShanker-geek commented on docs/diagrams/images/ArchitectureDiagram.png:
Starting the Developer Guide with high-level overview diagram is good for me quickly understand architecture. The amazing part that is looks way better than plantuml.

AjayShanker-geek commented on docs/diagrams/images/UiToCommand.png:
This is a minor issue, but it is difficult to see `1` at the head of the arrows.

AjayShanker-geek commented on docs/diagrams/images/RecurringExpenseLoadFromCsvSequence.png:
Good effort on the sequence diagrams. Two minor issues: firstly will be the conditions overlapping with activation bar makes it hard to read. Secondly, should the function called be closer to the head of the arrow?

AjayShanker-geek reviewed (commented):
Overall, your groups UG makes me worried for my team UG. Amazing work!

Progresst-8 committed:
Update UG FAQ

Progresst-8 committed:
Update help command with set spending limits feature

KHsienKit committed:
Merge pull request #104 from Progresst-8/UpdateCategoryDetails

Update UG and Help Command with Set feature

KHsienKit committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W12-3/tp

KHsienKit committed:
Merge branch 'master' into Edit-Command

AndrewNguyen4 committed:
Merge pull request #100 from KHsienKit/DesignGuide-RecurringExpense

Update Design Guide

KHsienKit committed:
Merge pull request #101 from shyaamald/branch-docs

Update UserGuide, HelpCommand and PPP

KHsienKit committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W12-3/tp

KHsienKit committed:
Merge pull request #102 from KHsienKit/Edit-Command

Update edit command

KHsienKit committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W12-3/tp

Progresst-8 committed:
Reformat method names in DG

Progresst-8 committed:
Update class UML to exclude constructors

Progresst-8 committed:
Reformat method names

Progresst-8 committed:
Add explanation for 2 add methods

Progresst-8 committed:
Remove mention of month in command table for clarity

Progresst-8 committed:
Update date and time handling section

Progresst-8 committed:
Add design consideration for dateutils

Progresst-8 committed:
Add explanation to checkLimitOf to method table

Progresst-8 committed:
Comment out attributes & methods that aren't planed to be discuss in DG

AndrewNguyen4 committed:
Add Visualization

MVP of Visualization command

AndrewNguyen4 committed:
Make minor code changes

AndrewNguyen4 committed:
Update Visualizer

- Remove unused imports & wildcard imports
- Visualizer can now work for time ranges up to three years

AndrewNguyen4 committed:
Fix CI

AndrewNguyen4 committed:
Fix CI

AndrewNguyen4 committed:
Update Architecture Diagram

Update PowerPoint source file for Architecture Diagram

Progresst-8 committed:
Update class UML

Progresst-8 committed:
Make spacing consistent

Progresst-8 committed:
Remove full stops

KHsienKit committed:
Merge pull request #106 from AndrewNguyen4/Add-Visualization

Add Visualization

KHsienKit committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W12-3/tp

KHsienKit committed:
Update Class diagram for recurringExpense

KHsienKit committed:
Update user guide

KHsienKit committed:
Fix typos

KHsienKit committed:
Update RecurringExpense sequence diagram

KHsienKit committed:
Update RecurringExpense sequence diagram png

KHsienKit committed:
Fix bug where user can add negative prices

KHsienKit committed:
Fix bug where price is now displayed with 2dp

shyaamald committed:
Update DG

shyaamald committed:
Fix text alignment in DG

shyaamald committed:
Merge pull request #112 from shyaamald/branch-docs

Update DG

AndrewNguyen4 committed:
Merge pull request #111 from KHsienKit/Fix-Price-Bug

Fix Price bug

AndrewNguyen4 committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W12-3/tp

Progresst-8 committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W12-3/tp
into UpdateDevGuideForMilestone2.0

# Conflicts:
#	docs/DeveloperGuide.md

shyaamald committed:
Implement help feature for specific methods

shyaamald committed:
Merge branch 'master' into branch_help_command

shyaamald committed:
Commit MyLogFile

shyaamald committed:
Merge branch 'branch_help_command' of https://github.com/shyaamald/tp into branch_help_command

# Conflicts:
#	MyLogFile.log

shyaamald committed:
Fix CI

AndrewNguyen4 committed:
Merge pull request #113 from shyaamald/branch_help_command

Implement Help feature for specific methods

AndrewNguyen4 committed:
Update DG

Reformatting the current content of DG for better cohesion and consistency.

AndrewNguyen4 committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W12-3/tp

AndrewNguyen4 committed:
Merge to Update-Documentation-2.0

AndrewNguyen4 committed:
Update DG

Add Visualize command to DG

AndrewNguyen4 committed:
Update UG & DG

- Slightly modified VisualizeCommand and Visualizer:
  - VisualizeCommand can now take no arguments, or take both /category and /time.
  - Visualizer: Rename variable: `totalExpenses` to `valueSeries` for clarity and consistency.
- Updated UG to reflect this change
- Added Visualizer to DG

AndrewNguyen4 committed:
Update Architecture Diagram

KHsienKit committed:
Merge pull request #114 from AndrewNguyen4/Update-Documentation-2.0

Update Documentation for Milestone 2.0

KHsienKit committed:
Update constructor for recurring expense

KHsienKit committed:
Merge branch 'master' into Add-JUnit-RecurringExpense

KHsienKit committed:
Update list command for recurring expense

Progresst-8 committed:
Update implementation of CategoryTracker.editCategory

KHsienKit committed:
Fix list recurring expense bug

Fix list recurring expense bug where it did not filter by date properly

Progresst-8 committed:
Update DG

Progresst-8 committed:
Fix logic for loading Category limits from CSV file

Progresst-8 committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W12-3/tp
into UpdateDevGuideForMilestone2.0

# Conflicts:
#	docs/DeveloperGuide.md

Progresst-8 committed:
Update DG

Progresst-8 committed:
Fix table

Progresst-8 committed:
Fix code to pass checkstyleMain

KHsienKit committed:
Fix Ui bug of displaying of expenses

Dates of expenses were displayed in YYYY-MM-DD format instead of DD-MM-YYYY format

Progresst-8 committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W12-3/tp

Progresst-8 committed:
Merge branch 'master' into UpdateDevGuideForMilestone2.0

Progresst-8 committed:
Readd missing line to DG

KHsienKit committed:
Update help command

KHsienKit committed:
Merge pull request #105 from Progresst-8/UpdateDevGuideForMilestone2.0

Update DG again for Milestone 2.0

KHsienKit committed:
Update RecurringExpense class diagram

KHsienKit committed:
Merge branch 'master' into Add-JUnit-RecurringExpense

KHsienKit committed:
Fix edit bug where price was not specified

KHsienKit committed:
Fix addDailyExpense bug

KHsienKit committed:
Add JUnit tests

KHsienKit committed:
Clean up JUnit test code

KHsienKit committed:
Fix CI bug in HelpCommand

KHsienKit committed:
Fix delete bug for recurring expense bug

KHsienKit committed:
Add IO test

KHsienKit committed:
Dummy commit to force subsequent commit

KHsienKit committed:
Fix date change

KHsienKit committed:
Fix EXPECTED.TXT to pass CI

Github is ona different timezone so it calls DateUtils.getCurrentDate() differently and calls a different date

KHsienKit committed:
Fix typo for CI

KHsienKit committed:
Remove IO test

Hackin7 committed:
Merge pull request #116 from KHsienKit/Add-JUnit-RecurringExpense

Add JUnit test for recurring expense

Progresst-8 committed:
Update Javadoc for facade methods

Progresst-8 committed:
Add UG FAQ

Hackin7 committed:
Add Javadocs

Progresst-8 committed:
Update comment according to KHsienKit

Progresst-8 committed:
Add link to Visualizer

Progresst-8 committed:
Update DG

Progresst-8 committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W12-3/tp
into CorrectDGforM2.0

# Conflicts:
#	docs/DeveloperGuide.md

KHsienKit committed:
Merge pull request #118 from Progresst-8/CorrectDGforM2.0

Update documentation before Test Run

AndrewNguyen4 committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W12-3/tp

Hackin7 committed:
Fix crash on loading from invalid float

Hackin7 committed:
Fix CI

Hackin7 committed:
Fix error message visual bug

Hackin7 committed:
Update UG on loading/ saving file format

Hackin7 committed:
Document parser edge cases

Hackin7 committed:
Add expected output to UG

Hackin7 committed:
Improve output on empty listing

Hackin7 committed:
Move Storage to its own section

Hackin7 committed:
Clarify set command

Hackin7 committed:
Fix delete recur in UG

KHsienKit committed:
Merge pull request #169 from Hackin7/Fix-Bugs

Bulk Big fixing from PE Dry Run

AndrewNguyen4 committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W12-3/tp

Hackin7 committed:
Refactor ArgumentsMap

Hackin7 committed:
Refactor AddCommand error handling

Progresst-8 committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W12-3/tp

Progresst-8 committed:
Clear tracker before loading from save file

Hackin7 committed:
Refactor EditCommand for command checking

Hackin7 committed:
Fix CI

Hackin7 committed:
Fix tests

Progresst-8 committed:
Fix bug that prints warning/alert message twice

LoadCommand calls load methods from ExpenseList and CategoryFacade:

ExpenseList.loadToCsv() calls CategoryFacade.addCategory()
CategoryFacade.addCategory() calls CategoryTracker.addCategory()

CategoryFacade.loadCategoryInfo() calls CategoryFacade.loadFromCsv()
CategoryFacade.loadFromCsv() calls CategoryStorage.trackCategoriesOf()
CategoryStorage.trackCategoriesOf() calls CategoryTracker.addCategory()

As a result, CategoryTracker.addCategory() is called twice.
For every expense in the save file, the price of that expense will be
added twice to the total of the category, causing the message to appear
twice

Progresst-8 committed:
Delete comment

Progresst-8 committed:
Add exception handling when curr or max expenditure is negative

Progresst-8 committed:
Add warning after setting a new spending limit

AndrewNguyen4 committed:
Fix issues #161 and #163

- Increase contrast for Architecture Diagram.
- Emphasise date limits of `visualize` in UG.

AndrewNguyen4 committed:
Add tests for ExpenseFilter

AndrewNguyen4 committed:
Update PPP - Minh Duc

AndrewNguyen4 committed:
Fix CI

Hackin7 committed:
Merge pull request #174 from AndrewNguyen4/Update-Documentation-2.1

Update Documentation for v2.1

KHsienKit committed:
Hide footboxes in DG Sequence diagrams

KHsienKit committed:
Add more information about recurring expense to UG

KHsienKit committed:
Fix CI

Progresst-8 committed:
Restructure EditCommand.execute() for clarity

KHsienKit committed:
Fix UG

KHsienKit committed:
Fix UG and help command for add recurring expense

KHsienKit committed:
Update help command

KHsienKit committed:
Add test methods for recurring expenses to DG

Progresst-8 committed:
Rewrote EditCommand execute method for clarity

Progresst-8 committed:
Format expenditure data to 2 decimal places for warnings/alerts

AndrewNguyen4 committed:
Fix CI

KHsienKit committed:
Fix load bug

Recurring expenses would readd normal expenses as lastAddedDate as not updated correctly

AndrewNguyen4 committed:
Merge pull request #173 from Progresst-8/fixIssue154

Add warning/alert when setting limits

AndrewNguyen4 committed:
Remove unused import

KHsienKit committed:
Merge pull request #175 from AndrewNguyen4/Add-Tests-2.1

Add ExpenseFilterTest

KHsienKit committed:
Merge pull request #171 from Progresst-8/fixIssue152

Fix issues relating to warnings/alerts when saving then loading

Progresst-8 committed:
Use Ui.displayMessage instead of system.out for printing stuff

KHsienKit committed:
Fix broken links in UG and DG

KHsienKit committed:
Fix DG broken link

KHsienKit committed:
Fix DG link bugs

KHsienKit committed:
Merge pull request #172 from Progresst-8/FixIssue150

Fix issues relating to warnings/alerts after edit

Progresst-8 committed:
Fix bug where exception thrown when editing category of an expense

This occurs when editing category of expense, and there are no more
expenses with the old category.

Before, the expense's category will get updated to newCategory.
Tracker is then edited, and attempts to decrement total of oldCategory.
To do so, it needs to call getCategoryDataOf first to find oldCategory.
However, key of oldCategory is no longer in the tracker because the
category of expense is newExpense and no longer oldExpense.
This causes getCategoryDataOf throws an exception.

KHsienKit committed:
Merge branch 'master' into V2.1

KHsienKit committed:
Update edit recurring expense UG

KHsienKit committed:
Update deleting recurring expense in UG and DG

KHsienKit committed:
Add authorship to help command

Hackin7 committed:
Merge branch 'master' into Fix-Bugs

Progresst-8 committed:
Add negative check for price attribute in Expense

KHsienKit committed:
Add error handling for negative price in save file

KHsienKit committed:
Merge pull request #170 from Hackin7/Fix-Bugs

Fix bugs round 2

KHsienKit committed:
Merge branch 'master' into V2.1

KHsienKit committed:
Fix CI

shyaamald committed:
Add javadoc comments, update PPP, and update stats command

Hackin7 committed:
Merge branch 'master' into Update-Documentation

Hackin7 committed:
Fix CI

shyaamald committed:
Merge branch 'master' into branch_testing

shyaamald committed:
Fix error in HelpCommand

shyaamald committed:
Fix Checkstyle

Progresst-8 committed:
Edit price checker from negative to nonpositive

Price now cannot be nonzero

KHsienKit committed:
Merge pull request #119 from Hackin7/Update-Documentation

Update Documentation

Hackin7 committed:
Update PPP

Hackin7 committed:
Add ArgumentsMap to UG

Hackin7 committed:
Update README.md

Hackin7 committed:
Merge pull request #176 from KHsienKit/V2.1

V2.1 Fixes

KHsienKit committed:
Merge pull request #181 from Hackin7/Update-Documentation

Update documentation

Hackin7 committed:
Update PPP to fit in page limit

Hackin7 committed:
Merge branch 'master' into branch_testing

KHsienKit committed:
Merge pull request #182 from Hackin7/Update-Documentation

Update PPP to fit in page limit

KHsienKit committed:
Update PPP

KHsienKit committed:
Update PPP

KHsienKit committed:
Merge branch 'master' into V2.1

KHsienKit committed:
Fix page break

Progresst-8 committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W12-3/tp
into FixIssue150

# Conflicts:
#	src/main/java/wheresmymoney/Expense.java
#	src/main/java/wheresmymoney/command/EditCommand.java

Progresst-8 committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W12-3/tp

Progresst-8 committed:
Merge branch 'FixIssue150'

shyaamald committed:
Fix author tags

shyaamald committed:
Merge branch 'branch_testing' of https://github.com/shyaamald/tp into branch_testing

Hackin7 committed:
Improve clarity of errors

Hackin7 committed:
Update error handling to be consistent

Hackin7 committed:
Merge pull request #184 from KHsienKit/V2.1

V2.1

Hackin7 committed:
Remove unused imports

shyaamald committed:
Fix author tag

Progresst-8 committed:
Update class UML diagram for category package

KHsienKit committed:
Merge pull request #180 from shyaamald/branch_testing

Add javadoc comments, update PPP, and update stats command

Hackin7 committed:
Update IO test

Hackin7 committed:
Fix bugs in UG

- Fix labelling to be more accurate
- Remove extra column from command summary
- Update Table of Contents
- Add disclaimer about numeric values

Hackin7 committed:
Add page break to UG

Hackin7 committed:
Add extra page break

AndrewNguyen4 committed:
Update PPP - Minh Duc

Progresst-8 committed:
Add sequence UML diagrams for 6 methods in CategoryFacade class

Progresst-8 committed:
Update UG contents page with Notes section

Hackin7 committed:
Add warning for exceeding limit

Hackin7 committed:
Add spacing in HelpCommand

Progresst-8 committed:
Update PPP

Hackin7 committed:
Update jar file name

Progresst-8 committed:
Update category sequence UMLs for Add, Delete, Edit

Progresst-8 committed:
Replace CategoryFacade method table with sequence diagrams

Hackin7 committed:
Update JavaDocs

Progresst-8 committed:
Update DG and PPP

Hackin7 committed:
Update test with stats

Progresst-8 committed:
Update PPP

Hackin7 committed:
Rename UI to Ui

Hackin7 committed:
Update docs for getRequiredLimit

Progresst-8 committed:
Update PPP

Progresst-8 committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W12-3/tp into updatePPPandUGandDG

Hackin7 committed:
Add Text UI Testing to DG

Progresst-8 committed:
Add javadoc for Expense constructors

Hackin7 committed:
Add Design & Implementation to ToC

Hackin7 committed:
Fix class name

Progresst-8 committed:
Update javadocs

Hackin7 committed:
Update class names to code style

Hackin7 committed:
Merge pull request #186 from AndrewNguyen4/Update-Documentation-2.1

Update PPP - Minh Duc

Hackin7 committed:
Update DG class diagram with ArgumentsMap

Hackin7 committed:
Add colon

Hackin7 committed:
Merge pull request #185 from Progresst-8/updatePPPandUGandDG

Update Documentation for Milestone 2.1 (PPP, UG, DG)

Hackin7 committed:
Merge branch 'master' into Fix-Bugs

Hackin7 committed:
Fix CI

Progresst-8 committed:
Update DG with correct link to category package class UML.

Add information about CategoryStorage

Hackin7 committed:
Update stats command

Hackin7 committed:
Fix text-ui-test

KHsienKit committed:
Merge pull request #187 from Progresst-8/updatePPPandUGandDG

Correct link to category class UML diagram

Hackin7 committed:
Fix text-ui-test ubuntu

Hackin7 committed:
Merge pull request #2 from AY2425S1-CS2113-W12-3/master

Update

KHsienKit committed:
Merge pull request #183 from Hackin7/Fix-Bugs

Final round of bug fixes

shyaamald committed:
Add calculations section to DG

shyaamald committed:
Merge branch 'master' into branch_testing

shyaamald committed:
Add calculations to ToC

Hackin7 committed:
Merge pull request #188 from shyaamald/branch_testing

Add calculations section to DG

Progresst-8 committed:
Edit my name to my full name in AboutUs

Update PPP details

Hackin7 committed:
Merge pull request #189 from Progresst-8/updatePPPandUGandDG

Update documentation

KHsienKit committed:
Fix typo in DG

Progresst-8 committed:
Update classUML with correct parameters

Progresst-8 committed:
Add png

Ignore the png in the commit before, that was the wrong png

Progresst-8 committed:
Simplify sequence diagram

Progresst-8 committed:
Update decorator capitalization indicating class level

Progresst-8 committed:
Update implementation description for CategoryFacade

KHsienKit committed:
Merge pull request #192 from Progresst-8/updatePPPandUGandDG

Update DG

Progresst-8 committed:
Merge pull request #191 from KHsienKit/V2.1

Fix typo in DG

Progresst-8 committed:
Correct UMLs

Progresst-8 committed:
Add isRecur

Progresst-8 committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W12-3/tp
into updatePPPandUGandDG

# Conflicts:
#	docs/diagrams/images/UiToCommand.png
#	docs/diagrams/plantuml/UiToCommand.puml

Progresst-8 committed:
Update UMLs

KHsienKit committed:
Merge pull request #193 from Progresst-8/updatePPPandUGandDG

Correct UMLs

shyaamald committed:
Update PPP and fix wrong file name

AndrewNguyen4 committed:
Merge pull request #194 from shyaamald/branch_testing

Update PPP and fix wrong file name
